### PR TITLE
Stack effect airflow

### PR DESCRIPTION
**Pull request by Mathadon**
@jelgerjansen I have made some code cleanups. 

Can you test whether the 2-port implementation of PPD12 and TwinHouse works in Dymola?

I.e. IDEAS/Examples/TwinHouses/BuildingO5_Exp1_2Port.mo  and IDEAS/Examples/PPD12/StructureExt_2Port.mo

@kldjonge fyi

### CODE REVIEW

**Comment by jelgerjansen** [comment]
Both models simulate, but there are some (translation) warnings for both models. I'll send you the details via e-mail.

**Comment by jelgerjansen** [comment]
Dag Filip

Hieronder de output van de simulaties.

Groetjes
Jelger

BuildingO5_Exp1_2Port.mo

  *   Translation

      [cid:d43ba4cb-c2f1-4a50-910c-b0d0760bdc55]
      [cid:80d716ef-0ca8-423c-ba6f-b7408edc6b98]

  *   Simulation

     dymosim started
... "IDEAS.Examples.TwinHouses.BuildingO5_Exp1_2Port" simulating
... "dsin.txt" loading (dymosim input file)
... loading "data" from "C:/Users/u0132350/Documents/GIT/IDEAS/IDEAS/Resources/measurements/validationdataO5Exp1.txt"
... loading "data" from "C:/Users/u0132350/Documents/GIT/IDEAS/IDEAS/Resources/measurements/BCTwinHouseO5.txt"
... loading "data" from "C:/Users/u0132350/Documents/GIT/IDEAS/IDEAS/Resources/measurements/MeasurementTwinHouseO5.txt"
... "BuildingO5_Exp1_2Port.mat" creating (simulation result file)

Integration started at T = 15000000 using integration method DASSL
(DAE multi-step solver (dassl/dasslrt of Petzold modified by Dassault Systemes))

Integration terminated successfully at T = 23587200
   CPU-time for integration                : 111 seconds
   CPU-time for one grid interval          : 11.6 milliseconds
   CPU-time for initialization             : 0.352 seconds
   Number of result points                 : 9954
   Number of grid points                   : 9543
   Number of accepted steps                : 52601
   Number of f-evaluations (dynamics)      : 112458
   Number of crossing function evaluations : 63581
   Number of Jacobian-evaluations          : 8444
   Number of model time events             : 5
   Number of input time events             : 0
   Number of state events                  : 202
   Number of step events                   : 0
   Minimum integration stepsize            : 0.000214
   Maximum integration stepsize            : 3.55e+04
   Maximum integration order               : 5
Calling terminal section
... "dsfinal.txt" creating (final states)

SUCCESSFUL simulation of IDEAS.Examples.TwinHouses.BuildingO5_Exp1_2Port

PPD12/StructureExt_2Port.mo

  *   Translation

[cid:8bca236f-69d2-4a6a-abe5-00caa8ff55c2]
[cid:1e8dde2e-7d14-4c73-b465-619ceb8fe0f3]


  *   Simulation

dymosim started

... "IDEAS.Examples.PPD12.StructureExt_2Port" simulating

... "dsin.txt" loading (dymosim input file)

Warning: The following was detected at time: 0

  In StructureExt_2Port.living.winC: You may have intended to model a frame since the parameter 'frac' is larger than zero. However, no frame type is configured such that no frame will be modelled. This may be a mistake. Set frac=0 to avoid this warning if this is intentional.

  Failed condition: false


Warning: The following was detected at time: 0

  In StructureExt_2Port.Porch.winA: You may have intended to model a frame since the parameter 'frac' is larger than zero. However, no frame type is configured such that no frame will be modelled. This may be a mistake. Set frac=0 to avoid this warning if this is intentional.

  Failed condition: false


Warning: The following was detected at time: 0

  In StructureExt_2Port.Porch.winCei: You may have intended to model a frame since the parameter 'frac' is larger than zero. However, no frame type is configured such that no frame will be modelled. This may be a mistake. Set frac=0 to avoid this warning if this is intentional.

  Failed condition: false


Warning: The following was detected at time: 0

  In StructureExt_2Port.bedRoom1.winC: You may have intended to model a frame since the parameter 'frac' is larger than zero. However, no frame type is configured such that no frame will be modelled. This may be a mistake. Set frac=0 to avoid this warning if this is intentional.

  Failed condition: false


Warning: The following was detected at time: 0

  In StructureExt_2Port.bathRoom.winA: You may have intended to model a frame since the parameter 'frac' is larger than zero. However, no frame type is configured such that no frame will be modelled. This may be a mistake. Set frac=0 to avoid this warning if this is intentional.

  Failed condition: false


Warning: The following was detected at time: 0

  In StructureExt_2Port.stairWay.winA: You may have intended to model a frame since the parameter 'frac' is larger than zero. However, no frame type is configured such that no frame will be modelled. This may be a mistake. Set frac=0 to avoid this warning if this is intentional.

  Failed condition: false


Warning: The following was detected at time: 0

  In StructureExt_2Port.bedRoom2.winC: You may have intended to model a frame since the parameter 'frac' is larger than zero. However, no frame type is configured such that no frame will be modelled. This may be a mistake. Set frac=0 to avoid this warning if this is intentional.

  Failed condition: false


Warning: The following was detected at time: 0

  In StructureExt_2Port.bedRoom3.winA: You may have intended to model a frame since the parameter 'frac' is larger than zero. However, no frame type is configured such that no frame will be modelled. This may be a mistake. Set frac=0 to avoid this warning if this is intentional.

  Failed condition: false


Warning: The following was detected at time: 0

  Could not find suitable correlation for air cavity! Please

      change the inclination to wall, ceiling or floor or remove the air layer.

  Failed condition: false


Warning: The following was detected at time: 0

  Could not find suitable correlation for air cavity! Please

      change the inclination to wall, ceiling or floor or remove the air layer.

  Failed condition: false


Warning: The following was detected at time: 0

  Could not find suitable correlation for air cavity! Please

      change the inclination to wall, ceiling or floor or remove the air layer.

  Failed condition: false


Warning: The following was detected at time: 0

  In StructureExt_2Port.stairWay.intFlo: Cavities are only supported for vertical walls, but inc=3.14159. The model is not accurate.

  Failed condition: false


Warning: The following was detected at time: 0

  In StructureExt_2Port.bedRoom3.intFlo: Cavities are only supported for vertical walls, but inc=3.14159. The model is not accurate.

  Failed condition: false


Warning: The following was detected at time: 0

  Could not find suitable correlation for air cavity! Please

      change the inclination to wall, ceiling or floor or remove the air layer.

  Failed condition: false


Warning: The following was detected at time: 0

  Could not find suitable correlation for air cavity! Please

      change the inclination to wall, ceiling or floor or remove the air layer.

  Failed condition: false


... "StructureExt_2Port.mat" creating (simulation result file)

NOTE: Adjusting output interval

 from 100 to 105 which is closest multiple of fixed step-size


Integration started at T = 0 using integration method EULER

(One-step solver with fixed stepsize and order 1)


Integration terminated successfully at T = 500010

   CPU-time for integration                : 25.1 seconds

   CPU-time for one grid interval          : 5.26 milliseconds

   CPU-time for initialization             : 0.115 seconds

   Number of result points                 : 4828

   Number of grid points                   : 4763

   Number of accepted steps                : 33334

   Number of f-evaluations (dynamics)      : 33334

   Number of crossing function evaluations : 33370

   Number of Jacobian-evaluations          : 0

   Number of model time events             : 0

   Number of input time events             : 0

   Number of state events                  : 35

   Number of step events                   : 0

   Minimum integration stepsize            : 15

   Maximum integration stepsize            : 15

   Maximum integration order               : 1

Calling terminal section

... "dsfinal.txt" creating (final states)


SUCCESSFUL simulation of IDEAS.Examples.PPD12.StructureExt_2Port

________________________________
Van: Filip Jorissen ***@***.***>
Verzonden: donderdag 22 juni 2023 22:22
Aan: open-ideas/IDEAS ***@***.***>
CC: Jelger Jansen ***@***.***>; Mention ***@***.***>
Onderwerp: [open-ideas/IDEAS] Stack effect airflow (PR #1322)


@jelgerjansen<https://github.com/jelgerjansen> I have made some code cleanups.

Can you test whether the 2-port implementation of PPD12 and TwinHouse works in Dymola?

I.e. IDEAS/Examples/TwinHouses/BuildingO5_Exp1_2Port.mo and IDEAS/Examples/PPD12/StructureExt_2Port.mo

@kldjonge<https://github.com/kldjonge> fyi

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/open-ideas/IDEAS/pull/1322

Commit Summary

  *   5be2d6d<https://github.com/open-ideas/IDEAS/pull/1322/commits/5be2d6d477d799248375dcf2d7ca3084da188f56> extending ReversibleDensityColumn.mo from MediumColumn and moved from baseclasses folder upwards
  *   0643e39<https://github.com/open-ideas/IDEAS/pull/1322/commits/0643e395d9d04b071d9db0c93d35c9deae776051> documentation and renamed Use_custom_Cs to use_custom_Cs
  *   5e4877a<https://github.com/open-ideas/IDEAS/pull/1322/commits/5e4877ad4bbc7e3542815b1930f3d0082aeffcc2> refactoring in PartialSurface
  *   1eba0de<https://github.com/open-ideas/IDEAS/pull/1322/commits/1eba0de6717231d56969313591403cb1bfe1875d> cleaned icon
  *   a302cfe<https://github.com/open-ideas/IDEAS/pull/1322/commits/a302cfe1e8dfc035d19fcc05d5f16b617f81910d> removed density columns from 1 port mode in internal wall
  *   cde76d4<https://github.com/open-ideas/IDEAS/pull/1322/commits/cde76d4e3e5d0791e7a8f750a252a5142d40c7a5> HideResult annotation

File Changes

(10 files<https://github.com/open-ideas/IDEAS/pull/1322/files>)

  *   D IDEAS/Airflow/Multizone/BaseClasses/ReversibleDensityColumn.mo<https://github.com/open-ideas/IDEAS/pull/1322/files#diff-e1bd3bd4c43de617428405f33d39ee7f7d3e17be95b26acfcae32ee9c3e62d39> (145)
  *   M IDEAS/Airflow/Multizone/BaseClasses/package.order<https://github.com/open-ideas/IDEAS/pull/1322/files#diff-90b340ebce9657179b901c944bfa70ce5d395a18eb90b3911ec786a3dcde7d3f> (1)
  *   A IDEAS/Airflow/Multizone/ReversibleDensityColumn.mo<https://github.com/open-ideas/IDEAS/pull/1322/files#diff-e4b5e97d0dc348e31bfdcdbe8ffa49f01a0482343202d746fe87976d1e6145b8> (44)
  *   M IDEAS/Airflow/Multizone/package.order<https://github.com/open-ideas/IDEAS/pull/1322/files#diff-d68257e3874a34399d06782913d19e6884bee9227c9de9305cb1ce7b1096c089> (1)
  *   M IDEAS/Buildings/Components/Examples/RectangularZoneTemplateOnePort.mo<https://github.com/open-ideas/IDEAS/pull/1322/files#diff-f5991cc513012f623f7f50f59adb9c562ef87d6ca04d9e04a1d839f0c167fe69> (2)
  *   M IDEAS/Buildings/Components/Examples/RectangularZoneTemplateTwoPort.mo<https://github.com/open-ideas/IDEAS/pull/1322/files#diff-2fa9a1b069b356c6ff8e835ddfa7d45cecac6655239bf2d8829273f6a8f68839> (2)
  *   M IDEAS/Buildings/Components/Interfaces/PartialSurface.mo<https://github.com/open-ideas/IDEAS/pull/1322/files#diff-6f8b47855b7d08aad140bd9aa252d81a3912bd2cfe0cf7776ff48400a70cdea2> (36)
  *   M IDEAS/Buildings/Components/InternalWall.mo<https://github.com/open-ideas/IDEAS/pull/1322/files#diff-0c70b9dab6d82a30eecb0d2c285c065baaca30d02e59b33aebc27bb961583bd9> (19)
  *   M IDEAS/Buildings/Components/OuterWall.mo<https://github.com/open-ideas/IDEAS/pull/1322/files#diff-5b502b00976b911907e6c3fbbfc8baddd4ab4c5fba09d28fd71e844eec8381c4> (13)
  *   M IDEAS/Buildings/Components/Window.mo<https://github.com/open-ideas/IDEAS/pull/1322/files#diff-052636080a1802e44e06219e42eb663f365476771ff3352334b43b37d1b1bb25> (9)

Patch Links:

  *   https://github.com/open-ideas/IDEAS/pull/1322.patch
  *   https://github.com/open-ideas/IDEAS/pull/1322.diff

File: IDEAS/Buildings/Components/Interfaces/PartialSurface.mo [hunk 1]
```diff
@@ -207,45 +207,29 @@ model PowerLaw_q50_stack
...
+    connect(port_a, res1.port_a) annotation (Line(points={{-100,0},{-80,0},{
           -80,40},{-68,40}}, color={0,127,255}));
-  connect(No_stack_a.port_b, res1.port_a) annotation (Line(points={{-48,40},{-20,
-          40},{-20,0},{-10,0}}, color={0,127,255}));
-  connect(res1.port_b, No_stack_b.port_a) annotation (Line(points={{10,0},{20,0},
+    connect(res1.port_b, port_b) annotation (Line(points={{10,0},{20,0},
           {20,40},{50,40}}, color={0,127,255}));
-  connect(No_stack_b.port_b, port_b) annotation (Line(points={{70,40},{80,40},{80,
-          0},{100,0}}, color={0,127,255}));
+  end if;
```
Comment (jelgerjansen) [inline IDEAS/Buildings/Components/Interfaces/PartialSurface.mo:232]: Just for clarification: if use_stack=true, in which component are the connections (to port_a, port_b, res1.port_a, res1.port_b) then drawn?

Comment (Mathadon) [reply IDEAS/Buildings/Components/Interfaces/PartialSurface.mo:232]: It seems that this code has been removed already on the HEAD so I'm not sure, but it was removed :p

**Comment by jelgerjansen** [comment]
@kldjonge @Mathadon all unit tests passed correctly, so I propose that I already merge this into the `StackEffectAirFlow` branch and continue with #1324?
@Mathadon I asked to update the revision history, but since the changes in this PR are limited, you might think this is not needed?

**Pull request merged**